File: apps/web/src/types/task.ts
export type Status = "pendentes" | "concluidas";

export interface Task {
  id: string;
  title: string;
  description?: string;
  status: Status;
  liked: boolean;
  likes: number;
  tag?: string;
  tag_color?: string;
  created_at: string;
  updated_at: string;
}

export interface CreateTaskInput {
  title: string;
  description?: string;
  status: Status;
  tag?: string;
  tag_color?: string;
}

export interface UpdateTaskInput {
  title?: string;
  description?: string;
  status?: Status;
  tag?: string;
  tag_color?: string;
}

File: api/internal/model/task.go
package model

import "time"

type Status string

const (
	StatusPendentes  Status = "pendentes"
	StatusConcluidas Status = "concluidas"
)

type Task struct {
	ID          string    `json:"id"`
	Title       string    `json:"title"`
	Description string    `json:"description,omitempty"`
	Status      Status    `json:"status"`
	Liked       bool      `json:"liked"`
	Likes       int       `json:"likes"`
	Tag         string    `json:"tag,omitempty"`
	TagColor    string    `json:"tag_color,omitempty"`
	CreatedAt   time.Time `json:"created_at"`
	UpdatedAt   time.Time `json:"updated_at"`
}

type CreateTaskInput struct {
	Title       string `json:"title"`
	Description string `json:"description"`
	Status      Status `json:"status"`
	Tag         string `json:"tag"`
	TagColor    string `json:"tag_color"`
}

type UpdateTaskInput struct {
	Title       *string `json:"title"`
	Description *string `json:"description"`
	Status      *Status `json:"status"`
	Tag         *string `json:"tag"`
	TagColor    *string `json:"tag_color"`
}
